### ISSUE TITLE
Is there any consequences removing  todoLi.id = i  and the .id and the parseInt part of the handlers.deleteTodos(parseInt(clickedElement.parentNode.id) ? Since the event is delegated to the <ul> as the wrapper of the <li>

### ISSUE THREAD

**Issue by shellhatcher** [gordonmzhu/questions#177]
### What is the associated Watch and Code lesson? 

version 10 
Clicking delete should update todoList.todos and the DOM
https://watchandcode.com/courses/60264/lectures/1102313

## Include a live demo link so that I can run your code. 

     
      **todoLi.id = i;** **( I removed this one)**
      todoLi.textContent = todoWithTextCompletion;
      //2.append deleteButton to todoLi
      todoLi.appendChild(this.createDeleteButton());
      todosUl.appendChild(todoLi);
      }
  },
  createDeleteButton : function(){
    var deleteButton = document.createElement("button");
    deleteButton.textContent = "Delete";
    deleteButton.className = "deleteButton";
    return deleteButton;
    
  },
  setUpEventListeners : function(){
    var todosUl = document.querySelector("ul");
    todosUl.addEventListener("click", function(event){
    var clickedElement = event.target;
    if(clickedElement.className === "deleteButton"){
      **handlers.deleteTodos(parseInt(clickedElement.parentNode.id)** **I removed the .id and parseInt part**

**codesourse**
https://plnkr.co/edit/FAkV8XUoaCqdUD6WdY2P?p=preview

## Explain the exact steps I need to take to replicate the problem. 

According to your explanation on event delegation, made me thought I can remove:
1. todoLi.id = i;
2.  -parse int and .id -part from handlers.deleteTodos(parseInt(clickedElement.parentNode.id)
so it became **handlers.deleteTodos(clickedElement.parentNode)**

### What did you expect to happen? 

The delete button will still working correctly, since we delegate the event to the ul we don't need to access each id of the specific  li
**My question is, is this approach I did correct? And will it have any consequences/problems in the future?**

### What actually happened? 

The delete button works as I assumed.

### Describe what you've done so far to troubleshoot the problem. 

Write your response here (required).

**Comment by gordonmzhu** [comment]
I don't quite follow your question as I can't tell exactly what you changed.

What specific line numbers should I be looking at?

**Comment by shellhatcher** [comment]
I am sorry for causing confusion, and thank you for teaching me how to point on the problem directly. I'll remember this.

The changes I made are as follows:
**1. Line 141, your codes included # todoLi.id = i #  I commented that out.
2. Line 159, I commented out the code you wrote which meant to access the id of the <li>
159 //handlers.deleteTodos(parseInt(clickedElement.parentNode.id); This code is the one you thought us.

Line 160, replaced the original code that you wrote
160 handlers.deleteTodos(clickedElement.parentNode); This code is the one I made change of.
Here I removed the .id after parentNode, and parseInt()** 

I did that because from what you taught, the event will be delegated into the <ul>, hence I thought I don't need to access the id of each <li>, since all <li> are under the <ul>, and the one which listen to the event is <ul> instead of <li>

To be honest I don't know whether what I thought was correct or not, that's why I opened a question.
and if that wasn't true, would you please kindly explain why it could work despite the removal of the codes?

Pardon for my lack of inability to explain things concisely.

Thank you very much in advance.

**Comment by gordonmzhu** [comment]
Thanks for clarifying @shellhatcher. That's very helpful.

After doing the following, what do you expect to have at the end?

1. Add three items: "First", "Second", and "Third".
2. Hit the delete button next to "Second".

You'd expect to have "First" and "Third" left right? (because you supposedly deleted "second"). What do you get in your app?

**Comment by shellhatcher** [comment]
Thank you very much,Gordon for not giving the answer directly, to let me experience on my own.
I truly appreciate this method of teaching. 

I tried several times on the app, by trying out several possibilities, and  all those possibilities (codes) which remove the exact id gave the same result. It deleted the top most of the list disregard of the item we clicked.

**So the answer to my question would be, the consequences of removing the codes meant to access the particular id of a list results in incorrect deletion of the list when we try to remove the list without an order (i.e; deleting other than the top most list). The removal will always done from the top most list.**

Is it true? 

For example deleting list number two resulted in deletion of list number one, and deletion of the last list also resulted in the deletion of the foremost list. It will only work correctly, if the deletion is done in an order from top to bottom (from first, second, and third, and so on).

In conclusion, for me to get consistence correct deletion result, I need to include codes which work to give the particular id or position of a list. Without it, JS will do the deletion according to its order within the ul.

Once again thank you very much indeed.

**Comment by gordonmzhu** [comment]
Did you use the debugger to see why your code doesn't work? 

When you do so, think about what type of data the `deleteTodo` method expects versus what data you are giving it. 

Specifically, think about the difference between `clickedElement.parentNode.id` and `clickedElement.parentNode`.

**Comment by shellhatcher** [comment]
Thanks for the answer.



2019年2月12日(火) 9:29、Gordon Zhu さん（notifications@github.com）のメッセージ:

**Comment by gordonmzhu** [comment]
Yep no problem @shellhatcher. I think once you compare the those two things (`clickedElement.parentNode.id` and `clickedElement.parentNode`), and then think about what `deleteTodo` expects as an argument, you'll see what the problem is.

**Comment by shellhatcher** [comment]
Thank you@Gordonmzhu.

I failed to think in an order, thanks for bringing this into notice.

Had I thought in order it would be clear:D

To delete something the method used is splice(). It takes more than one
arguments--->and position is one of them---->Position means number, the
link had not had number----> so I need to give it number( by giving id
using the number of the array)-->find a way to call the number---->use
event method to know the order to get it

event.target.parentNode returns Dom element, and it is not a number, so
passing this data to splice() won't give the result I want

While event.target.parentNode.id returns the id in the form of number (
number in a string form, that's why parseInt() is needed)  that we gave
based on the iteration of the array. This data type match the data type to
be passed into the splice() method.

I think from now on I need paper to write down the step and or requirement
to try not to loose track.




2019年2月14日(木) 3:39、Gordon Zhu さん（notifications@github.com）のメッセージ:

**Comment by gordonmzhu** [comment]
Yep you got it!
